### PR TITLE
Refactor TalentAccount API

### PR DESCRIPTION
**Pull request by bruce-fs**

### CODE REVIEW

File: lib/hubble_observatory/request.rb [hunk 1]
```diff
@@ -45,8 +45,12 @@ def uri
       @uri ||= URI::HTTPS.build host: host, path: "/api/v1/#{@route}", query: URI.encode_www_form(@query_params)
     end
 
+    def expected_response?
+      response.is_a?(Net::HTTPSuccess) || response.is_a?(Net::HTTPUnprocessableEntity)
+    end
```
Comment (claudiofullscreen) [inline lib/hubble_observatory/request.rb]: @bruce-fs Where is this method called?

File: lib/hubble_observatory/request.rb [hunk 2]
```diff
@@ -45,8 +45,12 @@ def uri
...
+
     def response
-      Net::HTTP.start(uri.host, 443, use_ssl: true) do |http|
+      @response ||= Net::HTTP.start(uri.host, 443, use_ssl: true) do |http|
```
Comment (claudiofullscreen) [inline lib/hubble_observatory/request.rb]: @bruce-fs Is there any place in the code that would benefit from this memoization?

File: lib/hubble_observatory/talent_account.rb [hunk 3]
```diff
@@ -2,39 +2,52 @@ module HubbleObservatory
...
     end
 
-    def self.extract_attribute_from_data(data:, attribute:)
-      data.fetch(:data, {}).fetch(:attributes, {}).fetch(attribute, nil)
+    def self.extract_hubble_uuid_from_data(data:)
+      data.fetch(:data, {}).fetch(:attributes, {}).fetch(:hubble_uuid, nil)
     end
 
-    def self.extract_uuid_from_errors(data:)
+    def self.extract_hubble_uuid_from_errors(data:)
```
Comment (claudiofullscreen) [inline lib/hubble_observatory/talent_account.rb]: @bruce-fs In general, I have a harder time reading this new file than the older one.

Is there anything specific that you are trying to achieve?

File: lib/hubble_observatory/request.rb (hunk 1 above)
Comment (bruce-fs) [inline lib/hubble_observatory/request.rb]: updated..

File: lib/hubble_observatory/talent_account.rb (hunk 3 above)
Comment (bruce-fs) [inline lib/hubble_observatory/talent_account.rb]: Let's discuss in person @claudiofullscreen when you're able

Comment (bruce-fs) [inline lib/hubble_observatory/talent_account.rb]: fyi - the changing of the method name from extract_uuid to  extract_hubble_uuid was originally proposed by you if I recall correctly

**Comment by bruce-fs** [comment]
Closing in favor of https://github.com/Fullscreen/hubble_observatory/pull/29

**Pull request closed without merging**